### ISSUE TITLE
Can't sync with Anki 2.1.0

### ISSUE THREAD

**Issue by jdoe0** [dsnopek/anki-sync-server#60]
Looks like Anki changed the sync protocol again. Attempting to sync with the server results in 2 error dialogs.

The first dialog contains the following stack trace:
````
Caught exception:
  File "/usr/share/anki/aqt/sync.py", line 346, in run
    self.col.close(save=False)
  File "/usr/share/anki/anki/collection.py", line 156, in close
    self.db.execute("pragma journal_mode = delete")
  File "/usr/share/anki/anki/db.py", line 31, in execute
    res = self._db.execute(sql, a)
<class 'sqlite3.OperationalError'>: cannot change out of wal mode from within a transaction
````

The second error dialog contains the following message:
````
Syncing failed:
'list indices must be integers or slices, not str'
````

**Comment by colonelsmoothie** [comment]
You should check out this issue, which discusses a port to Python 3:

https://github.com/dsnopek/anki-sync-server/issues/38

tsudoko has a fork which currently works with the Anki 2.1:

https://github.com/tsudoko/anki-sync-server

**Comment by tsudoko** [comment]
> Looks like Anki changed the sync protocol again.

Not really, the version checking code is just a bit wrong and assumes that a version with a higher patch number is newer. That makes the server treat Anki 2.1.x as a pre-2.0.13 version which used a different protocol.

I was thinking about making a PR to fix the check, but it seems like the server doesn't support pre-2.0.27 Anki versions since merging #28 anyway, so dropping the legacy code altogether might make more sense. I'll try to test it in more detail later and submit a relevant fix.
